test/postgrex_test.exs

```diff
@@ -1,30 +1,34 @@
 defmodule PostgrexTest do
   use ExUnit.Case, async: false
-  import ExUnit.CaptureLog
 
-  test "start_link/2 sets search path" do
-    # valid argument
-    search_path = ["public", "extension"]
-    {:ok, conn} = Postgrex.start_link(database: "postgrex_test", search_path: search_path)
-    %{rows: [[result]]} = Postgrex.query!(conn, "show search_path", [])
-    assert result == Enum.join(search_path, ", ")
+  # This test fails due to a bug betweem Elixir and Erlang in earlier versions of Elixir.
+  if Version.match?(Version.parse!(System.version()), Version.parse_requirement!(">= 1.17.2")) do
+    import ExUnit.CaptureLog
 
-    # invalid argument
-    Process.flag(:trap_exit, true)
-    search_path = "public, extension"
+    test "start_link/2 sets search path" do
+      # valid argument
+      search_path = ["public", "extension"]
+      {:ok, conn} = Postgrex.start_link(database: "postgrex_test", search_path: search_path)
+      %{rows: [[result]]} = Postgrex.query!(conn, "show search_path", [])
+      assert result == Enum.join(search_path, ", ")
```

Comment:
We can probably separate the happy path into its own test so it won't be ignored on lower versions.

Reply:
Done! Good call.